### PR TITLE
feat: accept semver compatible postcss AST

### PR DESCRIPTION
**Pull request by jquense**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

This is a nice optimization css-laoder does, but seems hard to hit, given how little control the end user has to ensure they only have one particular postcss.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
 /*
   MIT License http://www.opensource.org/licenses/mit-license.php
   Author Tobias Koppers @sokra
 */
 import validateOptions from 'schema-utils';
-import postcss from 'postcss';
-import postcssPkg from 'postcss/package.json';
-
-import { getOptions, isUrlRequest } from 'loader-utils';
+import satisfies from 'semver/functions/satisfies';
```
Comment (evilebottnawi) [inline src/index.js]: Let's use public API, no need hardcore path to function, it can be changed in theory

Comment (jquense) [reply src/index.js]: this is public https://www.npmjs.com/package/semver The readme, points to this for smaller footprints

File: package.json [hunk 2]
```diff
@@ -54,7 +54,8 @@
     "postcss-modules-scope": "^2.1.1",
     "postcss-modules-values": "^3.0.0",
     "postcss-value-parser": "^4.0.2",
-    "schema-utils": "^2.6.0"
+    "schema-utils": "^2.6.0",
+    "semver": "^7.1.2"
```
Comment (evilebottnawi) [inline package.json]: https://github.com/npm/node-semver/blob/master/package.json#L36

I think we need to use lower version

Comment (jquense) [reply package.json]: I can try and find one. Ya'll know that node 8 is End of Life right?

Comment (evilebottnawi) [reply package.json]: We still support node@8 https://github.com/webpack-contrib/css-loader/blob/master/package.json#L16, In the near future there will be a release that will throw out support node@8, but right now it is breaking change, so we should avoid it

Comment (jquense) [reply package.json]: yup, just wanted to make sure ya'll knew.

**Pull request merged**